src/utils/mtev_intern.c

```diff
@@ -322,7 +325,8 @@ borrow_free_node(mtev_intern_pool_t *pool,
   /* If we can manage to borrow the requested len without needing
    * to move this node into another freeslot, we can just "borrow"
    * the bytes from be done. */
-  while((oldsize = ck_pr_load_64(&n->size)) - len >= l->lsize) {
+  while((oldsize = ck_pr_load_64(&n->size)) >= l->lsize + len) {
```

Comment:
I don't quite follow the changes here and below.. Why is `size >= lsize + len` better than `size - len >= lsize` ?

Reply:
they are unsigned types.  So A - B >= C could have very bad behavior if  B > A.. (underflow).

By adding, you guarantee that you'd have to overflow to break (which is impracticle given that lengths of allocations here are stored in a unsigned:23